common/v2/features/BroadcastTransaction/stateFactory.tsx

```diff
@@ -0,0 +1,148 @@
+import { useContext } from 'react';
+import { Transaction as EthTx } from 'ethereumjs-tx';
+import { toBuffer, addHexPrefix } from 'ethereumjs-util';
+import { parseTransaction, Transaction } from 'ethers/utils';
+import BN from 'bn.js';
+
+import { TUseStateReducerFactory, toChecksumAddressByChainId, fromTxReceiptObj } from 'v2/utils';
+import { ITxReceipt, ITxConfig, ISignedTx, NetworkId } from 'v2/types';
+import { DEFAULT_NETWORK } from 'v2/config';
+import { translateRaw } from 'v2/translations';
+import { NetworkContext, getAssetByUUID, AssetContext } from 'v2/services/Store';
+import { fromWei, ProviderHandler } from 'v2/services/EthService';
+import { ToastContext } from 'v2/features/Toasts';
+
+const broadcastTxInitialState: State = {
+  network: DEFAULT_NETWORK,
+  txReceipt: undefined,
+  txConfig: undefined,
+  signedTransaction: '',
+  transaction: undefined,
+  networkSelectError: ''
+};
+
+interface State {
+  network: NetworkId;
+  txConfig: ITxConfig | undefined;
+  txReceipt?: ITxReceipt;
+  signedTransaction: ISignedTx;
+  networkSelectError: string;
+  transaction: EthTx | undefined;
+}
+
+const BroadcastTxConfigFactory: TUseStateReducerFactory<State> = ({ state, setState }) => {
+  const { networks, getNetworkByChainId, getNetworkByName } = useContext(NetworkContext);
+  const { assets } = useContext(AssetContext);
+  const { displayToast, toastTemplates } = useContext(ToastContext);
+
+  const handleNetworkChanged = (network: NetworkId) => {
+    setState((prevState: State) => ({
+      ...prevState,
+      network
+    }));
+  };
+
+  const handleSendClicked = (cb: any) => {
```

Comment:
Except the first line, ` const signedTx = hexlify(payload);` (because the user-input transaction should already be in hex-format), you could extract the below out to a function and use it instead of re-implementing it: https://github.com/MyCryptoHQ/MyCrypto/blob/master/common/v2/features/SendAssets/stateFactory.tsx#L129-L175

A cursory glance tells me you may have also missed ERC20 transfers vs ETH transactions in your implementation.

Reply:
Extracted code into another util function. Reusing this function also fixes second screenshot.
>2. There seems to be a problem resolving address book entry on Confirm Panel: